Cockpit 200 (#268)
cockpit-project/cockpit-project.github.io: _posts/2019-08-07-cockpit-200.md

---
title: Cockpit 200
author: pitti
date: 2019-08-07
tags: cockpit linux
slug: cockpit-200
category: release
summary: Cockpit with new TLS proxy, firewall, and Machines improvements
comments: 'true'
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/). We release regularly. This is version Two Hundred, prepared for you live and in person from the Cockpit team sprint in Berlin!

### Machines: Type-ahead OS selection

> The VM creation dialog merged the OS vendor and product inputs into a shared "Operating System" selector. This is more keyboard user friendly and supports type-ahead searching.

Reviewer: ```suggestion
In the VM creation dialog, OS vendor and product dropdowns have been replaced by a singular "Operating System" selector. The new dropdown is keyboard-friendly and supports auto-completion.
```

![Machines typeahead OS selection](/images/machines-os-typeahead.png)

### Machines: Create LVM storage pools

> You can now create new VM storage pools on LVM:

Reviewer: ```suggestion
Storage pools using LVM can be created on the Machines page.
```

![Machines LVM storage pools](/images/machines-lvm-pool.png)

### Networking: Show included firewalld services

[firewalld 0.7.1](https://firewalld.org/2019/07/firewalld-0-7-1-release) introduces the concept of services that include other services. Cockpit's Firewall page shows these:

![Firewall included services](/images/firewall-included-services.png)

### Web server: Split out TLS handling

> This release introduces [cockpit-tls](https://cockpit-project.org/guide/latest/cockpit-tls.8.html), a TLS terminating HTTP proxy that runs in front of [cockpit-ws](https://cockpit-project.org/guide/latest/cockpit-ws.8.html). This allows us to move TLS termination out of cockpit-ws, as that is hard to audit and trust: That code is subject to external attacks through HTTP, does a lot of interpretation of data streams, uses a lot of external dependencies, and multiplexes sessions of different users in one process. Thus any vulnerability can easily lead to complete privilege escalation across all running and future sessions.

Reviewer: ```suggestion
This release introduces [`cockpit-tls`](https://cockpit-project.org/guide/latest/cockpit-tls.8.html), a TLS terminating HTTP proxy that runs in front of [cockpit-ws](https://cockpit-project.org/guide/latest/cockpit-ws.8.html).

Moving TLS termination out of `cockpit-ws` improves auditing and trust. The `cockpit-tls` proxy is subject to external attacks through HTTP, does a lot of interpretation of data streams, uses a lot of external dependencies, and multiplexes sessions of different users in one process.

This split provides a foundation for isolating user sessions in the future, to increase security hardening. It also forms the basis of future authentication methods, including the use of smart cards.
```

> Right now this does not have any extra features over using cockpit-ws directly (other than better isolation), but it provides a basis for future work like supporting smart card authentication.

Reviewer: ```suggestion
```

> This change is expected to be transparent to users. Please report issues if you encounter any page loading failures, stalls, or similar networking issues with this version!

Reviewer: ```suggestion
This change is expected to be transparent to users. _Please [report issues](https://github.com/cockpit-project/cockpit/issues/new?template=bug_report.md) if you encounter any page loading failures, stalls, or similar networking issues with this version!_
```

### Try it out

Cockpit 200 is available now:

 * [For your Linux system](https://cockpit-project.org/running.html)
 * [Source Tarball](https://github.com/cockpit-project/cockpit/releases/tag/200)
 * [Fedora 30](https://bodhi.fedoraproject.org/updates/FEDORA-2019-d73df27df4)

*[VM]: Virtual Machine
*[LVM]: Logical Volume Manager

> *[OS]: Operating System

Reviewer: ```suggestion
*[OS]: Operating System
*[TLS]: Transport Layer Security
*[HTTP]: Hypertext Transport Protocol
```

![Machines typeahead OS selection](/images/machines-os-typeahead.png)

### Machines: Create LVM storage pools

Storage pools using LVM can be created on the Machines page.

![Machines LVM storage pools](/images/machines-lvm-pool.png)

### Machines: Create virtual networks

[Virtual networks](https://libvirt.org/formatdomain.html#elementsNICSVirtual) can now be created on the Virtual Machines page. These are used for connecting VMs with each other and/or to the host, or isolating different groups of VMs from each other.

![Machines create network](/images/machines-create-network.png)

### Networking: Show included firewalld services

[firewalld 0.7.1](https://firewalld.org/2019/07/firewalld-0-7-1-release) introduces the concept of services that include other services. Cockpit's Firewall page shows these:

![Firewall included services](/images/firewall-included-services.png)

### Web server: Split out TLS handling

This release introduces [`cockpit-tls`](https://cockpit-project.org/guide/latest/cockpit-tls.8.html), a TLS terminating HTTP proxy that runs in front of [cockpit-ws](https://cockpit-project.org/guide/latest/cockpit-ws.8.html).

> Right now this does not have any extra features over using cockpit-ws directly (other than better isolation), but it provides a basis for future work like supporting smart card authentication.

Reviewer: ```suggestion
At the moment, `cockpit-tls` does not provide additional features over using `cockpit-ws` directly (other than better isolation), but it sets up a basis for future work, such as supporting smart card authentication.
```

> This change is expected to be transparent to users. Please [report issues](https://github.com/cockpit-project/cockpit/issues/new?template=bug_report.md) if you encounter any page loading failures, stalls, or similar networking issues with this version!

Reviewer: ```suggestion
This change is expected to be transparent to users. Please [report issues](https://github.com/cockpit-project/cockpit/issues/new?template=bug_report.md) if you encounter any page loading failures, stalls, or similar networking issues!
```

Reviewer: "with this version" is redundant
